client/src/Pages/User/Cart/index.js

```diff
@@ -101,20 +192,21 @@ function CartDetail () {
             <hr className='breakline' />
             <div className='total'>
               <p className='total__header'>Total</p>
-              <p>{total.format()}</p>
+              <p>{currency(totals.subtotal + totals.tax).format()}</p>
             </div>
             <hr className='breakline' />
           </div>
         </div>
 
         <div className='float-cart__footer'>
-          <div
+          <button
+            disabled={cartItems().length == 0 || pickupTime == null}
```

Comment:
I was able to manually enter an invalid time or even use the Now button even though the store should be closed. I could still use the buy button. You might want to either check if pickUp time is valid or change Now is ASAP and disable typing in the dropdown.